### PR TITLE
API 10421 duplicate create errors

### PR DESCRIPTION
**Pull request by bschofield-va**
New set of exceptions for requests rejected by vista, which include "attempt to create duplicate record"

Contains a re-write of `LhsGatewayErrorHandler`

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/R4Controllers.java [hunk 1]
```diff
@@ -14,14 +14,6 @@
 
 @Slf4j
 public class R4Controllers {
-  /** If there are any errors that can be collected from the response, throw a fatal error. */
-  public static void dieOnError(LhsLighthouseRpcGatewayResponse response) {
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/R4Controllers.java:18]: moved to `LhsGatewayErrorHandler` unchanged

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java [hunk 2]
```diff
@@ -241,19 +240,6 @@ OperationOutcome handleCannotUpdateResourceWithMismatchedIds(
     return responseFor("processing", e, request, emptyList(), true, detail);
   }
 
-  @ExceptionHandler({ResourceExceptions.CannotUpdateUnknownResource.class})
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java:244]: Given that updates can modify multiple files, I no longer feel good about returning a `422` when we detected a failed to update fileman file. For now, a `500` will returned.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandler.java [hunk 3]
```diff
@@ -1,112 +1,219 @@
...
 
+@SuppressWarnings("ClassCanBeRecord")
 @AllArgsConstructor(staticName = "of")
 public class LhsGatewayErrorHandler {
+  private static final List<ErrorDetector> ERROR_DETECTORS =
+      List.of(
+          ErrorDetection.detectInvalidField(),
+          ErrorDetection.detectUnknownField(),
+          ErrorDetection.detectRecordDoesNotExist(),
+          ErrorDetection.detectRecordAlreadyExists());
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandler.java:37]: Static import these?

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandlerTest.java [hunk 4]
```diff
@@ -20,20 +21,33 @@
 import org.junit.jupiter.params.provider.MethodSource;
 
 public class LhsGatewayErrorHandlerTest {
-  private static Stream<Arguments> lhsGatewayErrors() {
+
+  public static final String ANYTHING = "(?s).*";
```
Comment (jhulbert-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandlerTest.java:25]: Is the goal here to eventually check for specific exception messages?

Comment (bschofield-va) [reply vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandlerTest.java:25]: prolly not... just making sure they say _something_.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandler.java (hunk 3 above)
Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/LhsGatewayErrorHandler.java:37]: 🤷‍♂️

**Review by jhulbert-va — APPROVED** [review]

**Review by ilaflamme-va — APPROVED** [review]

**Pull request merged**